mkosi/__init__.py

```diff
@@ -2551,37 +2565,32 @@ def install_arch(args: CommandLineArguments, root: Path, do_run_build_script: bo
         # Disable depmod pacman hook as depmod is handled by kernel-install as well.
         hooks_dir.joinpath("60-depmod.hook").symlink_to("/dev/null")
 
-        hooks_dir.joinpath("90-mkosi-kernel-add.hook").write_text(
-            importlib.resources.read_text("mkosi.resources.arch", "90_kernel_add.hook")
+        write_resource(hooks_dir / "90-mkosi-kernel-add.hook", "mkosi.resources.arch", "90_kernel_add.hook")
+        write_resource(
+            scripts_dir / "mkosi-kernel-add",
+            "mkosi.resources.arch",
+            "kernel_add.sh",
+            executable=True,
         )
 
-        kernel_add_script = scripts_dir / "mkosi-kernel-add"
-        kernel_add_script.write_text(importlib.resources.read_text("mkosi.resources.arch", "kernel_add.sh"))
-        make_executable(kernel_add_script)
-
-        kernel_remove_hook = hooks_dir / "60-mkosi-kernel-remove.hook"
-        kernel_remove_hook.write_text(importlib.resources.read_text("mkosi.resources.arch", "60_kernel_remove.hook"))
-
-        kernel_remove_script = scripts_dir / "mkosi-kernel-remove"
-        kernel_remove_script.write_text(importlib.resources.read_text("mkosi.resources.arch", "kernel_remove.sh"))
-        make_executable(kernel_remove_script)
+        write_resource(hooks_dir / "60-mkosi-kernel-remove.hook", "mkosi.resources.arch", "60_kernel_remove.hook")
+        write_resource(
+            scripts_dir / "mkosi-kernel-remove",
+            "mkosi.resources.arch",
+            "kernel_remove.sh",
+            executable=True,
+        )
```

Comment:
This really should be formatted as 
```py
       write_resource(hooks_dir / "60-mkosi-kernel-remove.hook", "mkosi.resources.arch", "60_kernel_remove.hook")
       write_resource(scripts_dir / "mkosi-kernel-remove", "mkosi.resources.arch", "kernel_remove.sh",
                                       executable=True)
```
Black makes it seem like two completely diferent things. OTOH, the PEP-8-style approach makes it easy to do a spot differences and similarities.

Obviously the same applies below: for no good reason, half of the calls to write_resource are compact, and half are spread.

Reply:
I'm not quite sold, that the PEP-8 style makes it easier to spot differences, but I'm generally not fond of overly long lines and prefer the more vertical style, as I find it more declarative. But this is something we can argue for a long time, now it's in the style my Emacs produces without interference :)